src/ddsrt/src/events/kqueue/events_kqueue.c

```diff
@@ -309,21 +311,46 @@ int ddsrt_event_queue_add(ddsrt_event_queue_t* queue, ddsrt_event_t* evt)
   queued_event_t* qe = &(queue->newevents[queue->nnewevents++]);
   qe->status = EVENT_STATUS_UNREGISTERED;
   qe->external = evt;
-  EV_SET(&qe->internal, evt->data.socket.sock, EVFILT_READ, EV_ADD, 0, 0, 0);
+  qe->evt_type = evt->type;
```

Comment:
GitHub won't let me comment on lines that haven't changed ... but I do want to remark that scanning all events in `add` to check for duplicates does mean that adding _n_ entries is an O(_n_^2) operation, and that's not so nice.

I'd be willing to require that the entries must be unique.

Reply:
If we can drop the requirement for uniqueness checks of events in the event queue, then we can indeed drop the O(N) checks, resulting in O(N^2) complexity.
Removed the uniqueness checks from the code.